torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -520,6 +540,74 @@ struct CodeImpl {
     return constant_table_;
   }
 
+  void process_ops_for_mobile() {
```

Comment:
IMHO, this algorithm can be optimized in implementation. What we really need in the metadata table is a mapping from op type to number of arguments serialized. So we can implement it this way to achieve equivalent functionality as what's currently implemented:

Python-like pseudo code
```
op_to_num_args = {}
for op in graph:
    if op not in op_to_num_args:
           op_to_num_args[op] = 0

    for (i, input) in enumerate(op.args()):
        if input is necessary:
            op_to_num_args[op] = max(i + 1, op_to_num_args[op])

    return op_to_num_args
```

Make sure to use iterators more when implementing in C++ to save map lookups.

Reply:
sure. makes sense